Add `checkContextFunctionalPseudoClasses: []`  to `selector-max-id` (#7380)
stylelint/stylelint: lib/rules/selector-max-id/README.md

```css
a:has(#foo) {}
```

### `includeContextFunctionalPseudoClasses: ["/regex/", /regex/, "non-regex"]`

> This option will include selectors inside of the array elements.

Reviewer: ```suggestion
Include selectors inside of specified functional pseudo-classes in the check.

This option has a higher precedence than `ignoreContextFunctionalPseudoClasses`.
```

Given:

```json
[":--foo"]
```

The following patterns are considered problems:

> <!-- prettier-ignore -->
> ```css
> :--foo(#foo) {}
> ```
> 

Reviewer: ```suggestion
```

The example needs to relate to "2" primary option from the top of README.

<!-- prettier-ignore -->
```css
:--foo(#foo #bar #baz) {}
```

The following patterns are _not_ considered problems:

<!-- prettier-ignore -->
```css

> :--foo { --fizz: 1px;}

Reviewer: ```suggestion
:--foo() {}
```

Let's make this functional.

```

> <!-- prettier-ignore -->
> ```css
> :-webkit-any(#foo) {}
> ```
> 

Reviewer: ```suggestion
```

Let's removed as unrelated to positive examples.

> This option has a higher presedence than `ignoreContextFunctionalPseudoClasses`.
> 
> For example, given if maximum selectors allowed are 2 and `:not` is included in `ignoreContextFunctionalPseudoClasses` and `includeContextFunctionalPseudoClasses`:
> 
> ```json
> [':not', /^:(glob|loc)al/, ':webkit-any']
> ```
> 
> The following patterns are considered problems:
> 
> <!-- prettier-ignore -->
> ```css
> :webkit-any(#foo #bar #baz) {}
> ```
> 
> <!-- prettier-ignore -->
> ```css
> a:not(#foo #bar #bazz) {}
> ```
> 
> The following patterns are _not_ considered problems:
> 
> <!-- prettier-ignore -->
> ```css
> :local { foo: 1px; } {}
> ```
> 
> <!-- prettier-ignore -->
> ```css
> :-webkit-any(#foo #bar) {}
> ```
> 
> <!-- prettier-ignore -->
> ```css
> #foo :global(#bazz #fizz) :local(#bar) {}
> ```

Reviewer: ```suggestion
```

We tend to keep our READMEs simple and only show the most common use cases.
